src/lib/items/hooks/useItem.js

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState } from 'react';
+import { getProductById } from '../../../pages/api/Items';
+
+/**
+ *
+ * @param {number} productId
+ * @param {array} deps
+ * @returns [imgSrc, title, price, desc, tags, favoriteCount]
+ */
+export function useItem(productId = 0, deps = []) {
+  const [imgSrc, setImgSrc] = useState('');
+  const [title, setTitle] = useState('');
+  const [price, setPrice] = useState(0);
+  const [desc, setDesc] = useState('');
+  const [tags, setTags] = useState([]);
+  const [isFavorite, setIsFavorite] = useState(false);
+  const [favoriteCount, setFavoriteCount] = useState(0);
```

Comment:
데이터를 따로 선택하지마시고, 하나의 객체로 관리해주면 훨씬 수월하실꺼에요 

```suggestion
const [item,setItem] = useState({
    imgSrc: "",
    title: "",
    price: 0,
    desc: "",
    tags: [],
    isFavorite: false,
    favoriteCount: 0,
  });
```